test/odbc/psqlodbc/xml.cpp

```diff
@@ -0,0 +1,230 @@
+#include "psqlodbc_tests_common.h"
+#include "../src/query_generator.h"
+
+const string BBF_TABLE_NAME = "master.dbo.xml_table_odbc_test";
+// For BBF Connection
+//   Cannot prepend database name when creating/dropping view
+//   Must prepend database name when selecting from view
+const string BBF_VIEW_NAME = "dbo.xml_view_odbc_test";
+const string PG_TABLE_NAME = "master_dbo.xml_table_odbc_test";
+const string PG_VIEW_NAME = "master_dbo.xml_view_odbc_test";
+const string COL1_NAME = "pk";
+const string COL2_NAME = "data";
+const string DATATYPE_NAME = "xml";
+const vector<pair<string, string>> TABLE_COLUMNS = {
+  {COL1_NAME, "INT PRIMARY KEY"},
+  {COL2_NAME, DATATYPE_NAME}
+};
+
+const string SIMPLE_XML = "<root>\
+<book>-148708948</book>\
+<leaf>simple</leaf>\
+<remain>shake</remain>\
+<football>nest</football>\
+<blanket>1275892738</blanket>\
+<try>-249153758.16992378</try>\
+</root>";
+
+const string NORMAL_XML = "<root>\
+<that>hot</that>\
+<anyone>1386369263.1890697</anyone>\
+<finest>believed</finest>\
+<song>\
+<stems>\
+<headed>1563641070.153514</headed>\
+<bring>-1175941901.956747</bring>\
+<stand>-494359546.04893684</stand>\
+<pole>774686962</pole>\
+<outer>production</outer>\
+<why>applied</why>\
+</stems>\
+<breeze>1269756220</breeze>\
+<cookies>544666062</cookies>\
+<yesterday>grade</yesterday>\
+<power>-663154896.7426605</power>\
+<topic>stomach</topic>\
+</song>\
+<fact>-1834471409</fact>\
+<repeat>-1980145795</repeat>\
+</root>";
+
+const string LARGE_XML = "<Root>\
+ <sys.tables name=\"tnullfixeddecimal0100\" object_id=\"4195065\" schema_id=\"1\" parent_object_id=\"0\" type=\"U \" type_desc=\"USER_TABLE\" create_date=\"2022-02-03T19:23:37.030\" modify_date=\"2022-02-03T19:23:37.030\" is_ms_shipped=\"0\" is_published=\"0\" is_schema_published=\"0\" lob_data_space_id=\"0\" max_column_id_used=\"5\" lock_on_bulk_load=\"0\" uses_ansi_nulls=\"1\" is_replicated=\"0\" has_replication_filter=\"0\" is_merge_published=\"0\" is_sync_tran_subscribed=\"0\" has_unchecked_assembly_data=\"0\" text_in_row_limit=\"0\" large_value_types_out_of_row=\"0\" is_tracked_by_cdc=\"0\" lock_escalation=\"0\" lock_escalation_desc=\"TABLE\" is_filetable=\"0\" is_memory_optimized=\"0\" durability=\"0\" durability_desc=\"SCHEMA_AND_DATA\" temporal_type=\"0\" temporal_type_desc=\"NON_TEMPORAL_TABLE\" is_remote_data_archive_enabled=\"0\" is_external=\"0\" is_node=\"0\" is_edge=\"0\" />\
```

Comment:
Nit: This is indented by 1 space. Can you fix indentation by indenting to 2 spaces?